client/components/package.json

```diff
@@ -0,0 +1,45 @@
+{
+	"name": "@automattic/simple-components",
+	"version": "0.1.0-alpha.6",
+	"description": "React components, as used on WordPress.com",
+	"scripts": {
+		"build": "npx babel --only ./index.js,./badge/,./button/,./main/,./notice/ --out-dir cjs .",
+		"prepublishOnly": "npm run build"
+	},
+	"repository": {
+		"type": "git",
+		"url": "git+https://github.com/automattic/wp-calypso.git"
+	},
+	"keywords": [
+		"wordpress",
+		"react",
+		"components"
+	],
+	"author": "Automattic",
+	"license": "GPL-2.0+",
+	"bugs": {
+		"url": "https://github.com/automattic/wp-calypso/issues"
+	},
+	"homepage": "https://github.com/Automattic/wp-calypso/blob/master/docs/components.md",
+	"main": "./cjs/index.js",
+	"esnext": "./index.js",
```

Comment:
Should we mirror the Gutenberg lerna config (build scripts etc...)?

We do publish two versions per module as well:

- `main` which corresponds to the transpiled version
- `module` which is a transpiled version but still using ES6 modules which guarantee tree shaking etc...

Is it a common usage to use an `esnext` property for untranspiled code? Also, how do we (package consumers) know what babel config or transpiling is needed with this field if it's not normalized?

Reply:
+1 on using `main` and `module` as you said. 

the latest i've seen on publishing es6+ is https://babeljs.io/blog/2018/06/26/on-consuming-and-publishing-es2015+-packages

Reply:
See `react-router` as an example:
https://unpkg.com/react-router@4.3.1/package.json

Reply:
> Should we mirror the Gutenberg lerna config (build scripts etc...)?

I didn't want to necessarily introduce build scripts yet since that seemed overblown for the little amount of transpilation we're doing at this point (YAGNI -- might add them later as needed tho).

> We do publish two versions per module as well:
>
> - `main` which corresponds to the transpiled version
> - `module` which is a transpiled version but still using ES6 modules which guarantee tree shaking etc...
>
> Is it a common usage to use an `esnext` property for untranspiled code? Also, how do we (package consumers) know what babel config or transpiling is needed with this field if it's not normalized?
 
I picked up `esnext` from the 2ality blog, see PR desc:

> http://2ality.com/2017/07/npm-packages-via-babel.html, and for more information, http://2ality.com/2017/04/transpiling-dependencies-babel.html and http://2ality.com/2017/06/pkg-esnext.html.

Granted, this is far from standardized, so I'm happy to reconsider.